### DISCUSSION TITLE
Allow varying culture for media types

### DISCUSSION THREAD

**Discussion by LennardF1989** [umbraco/Umbraco-CMS#13733] [Features and ideas]
We're running the situation of wanting this more and more. Would be great to be able to upload media multiple times under one node depending on the culture, just like content. Then have a media picker that can respect those cultures.

I see the following situations:
- Non-varying MediaPicker property on ContentType => Pick varying MediaType => Should take the appropriate culture from the Media-node.
- Non-varying MediaPicker property on ContentType => Pick non-Varying MediaType => Should use the same Media-file for all cultures.
- Varying MediaPicker property on ContentType =>Pick varying MediaType => Should take the appropriate culture from the Media-node.
- Varying MediaPicker property on ContentType =>Pick non-varying MediaType => Should use the Media-file for the current culture.

This will probably have all kinds of implications, but it's worth looking into this.

Has been requested before: https://github.com/umbraco/Umbraco-CMS/issues/4771

**Comment by deanleigh** [comment]
This is also the first step required for adding Alt attributes to images in the media library which is now very important
[https://github.com/umbraco/Umbraco-CMS/issues/8432#event-8253669282](https://github.com/umbraco/Umbraco-CMS/issues/8432#event-8253669282)

**Comment by cryothic** [comment]
I would be happy with just the abillity to add multi-lingual fields to a media-type.

As @deanleigh said, alt attributes are important. For SEO, and for Accessability. Everybody is pushing to descriptive alt-texts, but it's just  painfull to do in Umbraco. Unless you place your alt-texts with an image in a block-list or something. 

It's a shame this isn't even on the roadmap at the moment.

**Comment by enkelmedia** [comment]
This is something that we've been wanting for a long time as well. Our current workaround is to use a text-field property for each culture but it would be a lot nicer if media properties could "vary by culture" as well.

**Comment by DanDiplo** [comment]
This package by Dave is handy for adding multi-lingual text fields - https://github.com/dawoe/umbraco-multilanguage-textbox

**Reply by cryothic** [reply]
That looks good.
I assume it saves the data like regular multi-lingual content, as it just uses `.Value()`. This makes it even more frustrated this isn't in Umbraco by default.
I will keep this package in mind next time! Thanks.

**Comment by enkelmedia** [comment]
Thanks Dan! 

It’s not a difficult problem to solve for a individual build. The problem is rather around further proofing and facilitate a nice upgrade path that is not dependent on external packages for core features like localization.

**Comment by edihasler** [comment]
@iOvergaard @nul800sebastiaan is this meant to be implemented in the future?

**Reply by meixger** [reply]
> We need the media library to support variants. This is something we are currently tracking internally and unfortunately, something that has no fixed date set yet.

ref https://github.com/umbraco/Umbraco-CMS/discussions/13690#discussioncomment-5670847
